pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -42,13 +52,58 @@ func NewDRAPluginClient(pluginName string) (drapb.NodeClient, error) {
 	return existingPlugin, nil
 }
 
-func (p *plugin) NodePrepareResources(
+type Plugin struct {
+	ctx    context.Context
+	cancel func(cause error)
+
+	mutex                   sync.Mutex
+	conn                    *grpc.ClientConn
+	endpoint                string
+	highestSupportedVersion *utilversion.Version
+	clientTimeout           time.Duration
+}
```

Comment:
Why was this moved out of plugin.go?

Reply:
Or maybe the better question is -- should we move this here and rename this file to plugin.go and the original plugin.go to something else?

Reply:
Seems appropriate to rename this to plugin.go and the other file to registration.go

Reply:
Can we discuss that in https://github.com/kubernetes/kubernetes/pull/125163/files#diff-0b0c13558c483aa8d0584bdce871e43cd3b2f1b521208a10dd6515561b9acd15 and merge that PR first?

Reply:
Answered in https://github.com/kubernetes/kubernetes/pull/125163/files#r1665552886.